Update project2draft.qmd

Gwanook opened the pull request:
Rewrite the conclusion (test).

Gwanook committed:
Update project2draft.qmd

Rewrite the conclusion.

Erika026 reviewed (changes requested):
Need improve the conclusion.

Gwanook committed:
Update project2draft.qmd

The model fits all the variables, but the result is not really great.

Gwanook committed:
Update project2draft.qmd

Added the part of deleting outliers and changed the model again, but the formula and some analysis results have not been changed.

Erika026 commented on project2draft.qmd:
````diff
@@ -58,6 +58,26 @@ class(coffee1$Qualityclass)
 coffee1$Qualityclass <- as.factor(coffee1$Qualityclass)
 ```
 
+```{r}
````
**Appropriate steps were taken to remove the outlier.** 
- Improvement can be found directly in the boxplots; the shape of those plots performed much better than before. 

- Should be considered to keep in the main branch.

Erika026 commented on project2draft.qmd:
````diff
@@ -206,62 +224,34 @@ model2 <- glm(formula=Qualityclass~ aroma + flavor + acidity +
 ```
 
-Both model 5 and 6 (both contain 4 explanatory variables) have lower AIC value than Model 2, proving that the remove of that explanatory variable is effective. And Model 5 has the lower AIC value (569.67) thus we processed to further one variable removing based on Model 5.
+The AIC of model 3 is observed to increase following the removal of the variable harvested, as opposed to the expected decrease. Consequently, model 2 is deemed to be the most suitable, even though the variable harvested shows a p-value greater than 0.05.
````
**Inappropriate Model Selection**

- Should consider Model 3 rather than Model 2

When comparing the goodness of fit of Models:
Model 2: AIC = 545.34, BIC = 574.04 , with one variable not significant;
Model 3: AIC = 545.96 (a bit increase within two decimal places)
               BIC = 569.87 (obvious reduction) with all variables significant.
Recall that BIC penalizes more on the additional parameter, thus AIC aims to use it for prediction accuracy and BIC aim for finding true model.

Erika026 commented on project2draft.qmd:
```diff
@@ -295,31 +285,31 @@ mod.coef.logodds <- model7 |>
 #| label: fig-CIplot
 # 95% Confidence Interval Plot for Log-Odds.
-plot_model(model7, show.values = TRUE, transform = NULL,
+plot_model(model2, show.values = TRUE, transform = NULL,
```
Please **update all the 95% Confidence Interval plots (Log-Odds and Odds)** with appropriate models (Model 3 in this case).

Erika026 commented on project2draft.qmd:
````diff
@@ -295,31 +285,31 @@ mod.coef.logodds <- model7 |>
 ```
 
-For @tbl-logoddsCI, since all the intervals(\[-5.8,-3.2\] for aroma grade, \[-8.7,-5.4\] for flavor grade, \[-5.7,-3.1\] for acidity grade) don't contain zero thus indicates that the explanatory variables in model 7 are all significant. And the @fig-CIplotodd again confirmed the significant of the explanatiory variables visually.
+For @tbl-logoddsCI, since most of the intervals($[-6.0908,-3.2701]$ for aroma grade, $[-8.8598,-5.4496]$ for flavor grade, $[-5.4826,-2.8223]$ for acidity grade, $[-0.0013,-0.0003]$ for altitude_mean_meters grade, $[-0.2124,0.0203]$ for harvested grade), excluding the variable harvested, don't contain zero thus indicates that the explanatory variables in model 2 are significant. And the @fig-CIpl ...
````
Interpretation update is needed.

Erika026 commented on project2draft.qmd:
```diff
@@ -295,31 +285,31 @@ mod.coef.logodds <- model7 |>
 
 $$
-\frac{p}{1-p} = exp(\alpha + \beta_1 \cdot \textrm{aroma} +\beta_2 \cdot \textrm{flavor} + \beta_3 \cdot \textrm{acidicity})
+\frac{p}{1-p} = \exp(\alpha + \beta_1 \cdot \textrm{aroma} +\beta_2 \cdot \textrm{flavor} + \beta_3 \cdot \textrm{acidicity})
```
Required for updating the latex expression, one more variable needs to be added.

Erika026 commented on project2draft.qmd:
```diff
@@ -206,62 +224,34 @@ model2 <- glm(formula=Qualityclass~ aroma + flavor + acidity +
-             family = binomial(link = "logit"))
-
-summ(model7)
+#| echo: true
```
**Correct Flow of Model Fitting Needed**

- glm_model is the full model containing all explanatory variables, thus probably needs to be fitted at the first step.
-  Appropriate order suppose to be glm_model ->model -> Model 2  -> Model 3

Erika026 commented on project2draft.qmd:
```diff
@@ -362,13 +352,13 @@ The @fig-CIplotodd shows the 95% confidence intervals for three explanatory vari
 
 $$
-p = \frac{exp(\alpha + \beta_1 \cdot \textrm{aroma} +\beta_2 \cdot \textrm{flavor} + \beta_3 \cdot \textrm{acidicity})}{1 + exp(\alpha + \beta_1 \cdot \textrm{aroma} +\beta_2 \cdot \textrm{flavor} + \beta_3 \cdot \textrm{acidicity})}
+p = \frac{\exp(\alpha + \beta_1 \cdot \textrm{aroma} +\beta_2 \cdot \textrm{flavor} + \beta_3 \cdot \textrm{acidicity})}{1 + \exp(\alpha + \beta_1 \cdot \textrm{aroma} +\beta_2 \cdot \textrm{flavor} + \beta_3 \cdot \textrm{acidicity})}
```
Expression Latex Update Required

Erika026 commented on project2draft.qmd:
````diff
@@ -407,10 +397,32 @@ ggplot(data = coffee1, aes(x = acidity, y = probs.poor)) +
+```
+
+```{r}
+#| echo: true
````
The Probability of Quality Score **by Harvested** is no longer required when using Model 3.

Erika026 reviewed (changes requested):
**Improvements Required**

- Appropriate Model Selection (Model 3).
- Logical Flow for Model Fitting Required
- Update All the Plots(CI and Probability plots)

Gwanook committed:
Update project2draft.qmd

Regard the Model 3 as the final model. Interpretation and conclusion are left to you guys~ ☺️

Erika026 commented on project2draft.qmd:
````diff
@@ -235,7 +235,7 @@ model3 <- glm(formula = Qualityclass~ aroma + flavor + acidity +
 ```
 
-The AIC of model 3 is observed to increase following the removal of the variable harvested, as opposed to the expected decrease. Consequently, model 2 is deemed to be the most suitable, even though the variable harvested shows a p-value greater than 0.05.
+The AIC of model 3 is observed to increase a little following the removal of the variable harvested, but the BIC decrease as expected. Consequently, model 3 is deemed to be the most suitable.
````
All model selections and graphical summaries seem to be appropriate now.
The interpretation of keeping the full GLM model at the last part of model fitting is reasonable. Thus it's acceptable to have such flow.

Erika026 reviewed (approved):
- Appropriate changes of model selection and interpretation. 

- Able to merge to the main branch and process to the next stage.
